pkg/migrations/rename.go

```diff
@@ -120,42 +120,42 @@ func RenameDuplicatedColumn(ctx context.Context, conn db.DB, table *schema.Table
 		}
 	}
 
-	// Rename any `UNIQUE` indexes on the duplicated column and use them to
+	// Rename any indexes on the duplicated column and use unique indexes to
 	// create `UNIQUE` constraints.
-	for _, ui := range table.Indexes {
-		if !IsDuplicatedName(ui.Name) {
-			continue
-		}
-		if !ui.Unique {
+	for _, idx := range table.Indexes {
+		if !IsDuplicatedName(idx.Name) {
 			continue
 		}
 
-		if slices.Contains(ui.Columns, TemporaryName(column.Name)) {
-			// Rename the unique index to its original name
+		if slices.Contains(idx.Columns, TemporaryName(column.Name)) {
+			// Rename the index to its original name
 			renameIndexSQL := fmt.Sprintf(cRenameIndexSQL,
-				pq.QuoteIdentifier(ui.Name),
-				pq.QuoteIdentifier(StripDuplicationPrefix(ui.Name)),
+				pq.QuoteIdentifier(idx.Name),
+				pq.QuoteIdentifier(StripDuplicationPrefix(idx.Name)),
 			)
 
 			_, err = conn.ExecContext(ctx, renameIndexSQL)
 			if err != nil {
-				return fmt.Errorf("failed to rename unique index %q: %w", ui.Name, err)
+				return fmt.Errorf("failed to rename index %q: %w", idx.Name, err)
 			}
+		}
 
+		if idx.Unique {
```

Comment:
This means that `pgroll` will convert unique indexes on duplicated columns to unique constraints.

[This gist](https://gist.github.com/andrew-farries/1d2936989e9655ed69268c234a4b1d2c) shows the issue:

1. Create a table
2. Add a unique index to one of the table's fields
3. Set the field `NOT NULL`

After migration 2., the field has a unique index, but not a unique constraint.
After migration 3, the field has a unique constraint rather than an index.

I'm not sure how important this will be in practice, but we should probably avoid doing this as it's an unexpected change to the table.

Reply:
I fixed problem. Thank you for the migration files.